packages/app/src/providers/h5grove/h5grove-api.ts

```diff
@@ -149,37 +150,47 @@ export class H5GroveApi extends DataProviderApi {
 
   private async processEntityResponse(
     path: string,
-    response: H5GroveEntityResponse
-  ): Promise<Entity> {
-    const { name } = response;
+    response: H5GroveEntityResponse,
+    isChild: true
+  ): Promise<ChildEntity>;
+
+  private async processEntityResponse(
+    path: string,
+    response: H5GroveEntityResponse,
+    isChild?: false
+  ): Promise<ProvidedEntity>;
 
+  private async processEntityResponse(
+    path: string,
+    response: H5GroveEntityResponse,
+    isChild = false
+  ): Promise<ProvidedEntity | ChildEntity> {
+    const { name } = response;
     const baseEntity = { name, path };
 
     if (isGroupResponse(response)) {
-      const { children, attributes: attrsMetadata } = response;
+      const { children = [], attributes: attrsMetadata } = response;
       const attributes = await this.processAttrsMetadata(path, attrsMetadata);
       const baseGroup: Group = {
         ...baseEntity,
         kind: EntityKind.Group,
         attributes,
       };
 
-      if (!children) {
-        /* `/meta` stops at one nesting level
-         * (i.e. children of child groups are not returned) */
+      if (isChild) {
         return baseGroup;
       }
 
-      const groupWithChildren: GroupWithChildren = {
```

Comment:
Changing the return type to `Promise<ProvidedEntity | ChildEntity>` means we no longer need those intermediary variables (or any type casts).